src/Bicep.LangServer/Providers/LinterRulesProvider.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using Bicep.Core.Analyzers.Interfaces;
+
+namespace Bicep.LanguageServer.Providers
+{
+    public class LinterRulesProvider : ILinterRulesProvider
+    {
+        private readonly Lazy<ImmutableDictionary<string, string>> linterRulesLazy;
+
+        public LinterRulesProvider()
+        {
+            this.linterRulesLazy = new Lazy<ImmutableDictionary<string, string>>(() => GetLinterRulesInternal().ToImmutableDictionary());
+        }
+
+        private Dictionary<string, string> GetLinterRulesInternal()
```

Comment:
>GetLinterRulesInternal [](http://example.com/codeflow?start=43&length=22)

This code is very similar to the code the linter uses to construct instances of the rules to execute. You should refactor the linter to also use the linter rule provider and then this code here should just consume that list and just grab the codes and levels from them.